modules/drivers/sparksql/src/metabase/driver/sparksql.clj

```diff
@@ -29,13 +30,28 @@
   "Default alias for all source tables. (Not for source queries; those still use the default SQL QP alias of `source`.)"
   "t1")
 
-;; use `source-table-alias` for the source Table, e.g. `t1.field` instead of the normal `schema.table.field`
-(defmethod sql.qp/->honeysql [:sparksql (class Field)]
-  [driver field]
-  (binding [sql.qp/*table-alias* (or sql.qp/*table-alias* source-table-alias)]
-    ((get-method sql.qp/->honeysql [:hive-like (class Field)]) driver field)))
-
-(defmethod sql.qp/apply-top-level-clause [:sparksql :page] [_ _ honeysql-form {{:keys [items page]} :page}]
+(defmethod sql.qp/->honeysql [:sparksql :field]
+  [driver [_ _ {::params.substitution/keys [compiling-field-filter?]} :as field-clause]]
+  ;; use [[source-table-alias]] instead of the usual `schema.table` to qualify fields e.g. `t1.field` instead of the
+  ;; normal `schema.table.field`
+  (let [parent-method (get-method sql.qp/->honeysql [:hive-like :field])
+        field-clause  (mbql.u/update-field-options field-clause
+                                                   update ::add/source-table
```

Comment:
nit: move `::add/source-table` to next line?

Reply:
fixed